packages/nextjs/components/scaffold-eth/Input/UintInput.tsx

```diff
@@ -0,0 +1,64 @@
+import { BigNumber, ethers } from "ethers";
+import { useCallback, useEffect, useState } from "react";
+import { InputBase } from "./InputBase";
+import { CommonInputProps, NUMBER_REGEX, UintVariant } from "./utils";
+
+type UintInputProps = CommonInputProps<string | BigNumber> & {
+  variant?: UintVariant;
+};
+
+export const UintInput = ({ value, onChange, name, placeholder, variant = UintVariant.UINT256 }: UintInputProps) => {
+  const [inputError, setInputError] = useState(false);
+  const convertEtherToUint = useCallback(() => {
+    if (!value || value instanceof BigNumber) {
```

Comment:
But what if the user wants to pass some small `value` as `BigNumber` and he wants it to be a wei? For example for prefilling input. Or we don't want to consider this?

Reply:
Hmmm, fair point. Maybe it's less confusing for the user to not assume anything and let him multiply as many times as he wishes... :smiley: